kernelci/config/base.py

```diff
@@ -80,6 +80,24 @@ def match(self, **kw):
         """Return True if the given *kw* keywords match the filter."""
         raise NotImplementedError("Filter.match() is not implemented")
 
+    def combine(self, items):
+        """Try to avoid making a new filter if we can make a combined
+        filter matching both our existing data and the new items.
+
+        The *items* can be any data used to filter configurations.
+        Return True if we can combine, False otherwise.
+        """
+        return False
+
+
+def _merge_filter_lists(a, b):
+    """Merge the items for a Blocklist or Passlist."""
+    for key, value in b.items():
+        if key not in a:
+            a[key] = value
+        else:
+            a[key].extend(value)
```

Comment:
I think we can make use of `dict.setdefault()` here

```suggestion
        a.setdefault(key, []).extend(value)
```